block/algebraic/petsc/precond.py

```diff
@@ -11,55 +11,63 @@
 from time import time
 from dolfin import info
 
-
-class precond(block_base):
-    def __init__(self, A, prectype, pdes, nullspace, prefix, options, V=None, create_cb=None, defaults={}):
-        Ad = mat(A)
-
-        if nullspace:
-            from block.block_util import isscalar
-            ns = PETSc.NullSpace()
-            if isscalar(nullspace):
-                ns.create(constant=True)
-            else:
-                ns.create(constant=False, vectors=[v.down_cast().vec() for v in nullspace])
-            try:
-                Ad.setNearNullSpace(ns)
-            except:
-                info('failed to set near null space (not supported in petsc4py version)')
-
+class petsc_base(block_base):
```

Comment:
Some of the methods, e.g. `LU`, `Cholesky` miss the prefix argument in their `__init__` signature.

Reply:
I think it must be a problem introduced in PR #9, not this one? I'll look into it.

Reply:
Rebased on top of bugfix